lib/doorkeeper/config.rb

```diff
@@ -495,6 +499,22 @@ def option_defined?(name)
       instance_variable_defined?("@#{name}")
     end
 
+    # The method device_code_supported? checks, if the migration to enable device code grant
+    # was generated. Normally you use Doorkeeper::AccessGrant.device_code_supported?, since that
+    # just checks the column existing on Database.
+    # This here is the fallback for cases, where database could not be loaded or migrated already
+    # - as in some cases on routes.
+    # Sorry - I don't know, how to check this on Grape. But since lib/doorkeeper/rails/routes.rb
+    # is for rails exclusively - we currently do not use this method outside of Rails.
+
+    def device_code_supported?
+      return false unless defined?(::Rails)
```

Comment:
Looks very strange for me. Why do we need to search for migration instead of checking the model for the column? Like it is done for refresh token and some other config options :thinking:

Reply:
The other issues Dan will respond to. But this was workaround was done by me. I will need to reproduce the issue again. But I am quite sure, that there is no database connection yet, when routes are preloaded. So the big difference between refresh_token and device_code - are the extra routes, that refresh token does not need.

Reply:
We definitely need to think about other way to check for supported device_code. 

Also I'm thinking about can we move all this PR in some gem extension? I don't really like the idea of having all in one. It would be great if you can check the `doorkeeper-openid-connect` gem and we could move the logic to a standalone extension. I can create a repo for it and add you as a collaborators for it (if you want to).

Reply:
that sounds a good idea. We would like to work on it further.